src/frontend/app/features/cloud-foundry/services/cloud-foundry-endpoint.service.ts

```diff
@@ -138,20 +139,11 @@ export class CloudFoundryEndpointService {
     this.info$ = this.cfInfoEntityService.waitForEntity$;
 
     this.allApps$ = this.orgs$.pipe(
-      map(p => {
-        return p
-          .filter(o => !!o.entity.spaces)
-          .map(o => {
-            return o.entity.spaces.map(space => space.entity.apps || []);
-          });
-      }),
-      map(a => {
-        let flatArray = [];
-        a.forEach(
-          appsInSpace => (flatArray = flatArray.concat(...appsInSpace))
-        );
-        return flatArray;
-      })
+      map(orgs => [].concat(...orgs.map(org => org.entity.spaces))),
+      switchMap((spaces: APIResource<ISpace>[]) =>
+        combineLatest(spaces.map(space => this.store.select(selectEntity(spaceSchemaKey, space.metadata.guid))))
```

Comment:
This step isn't required. The space entities in the org entity should be up to date.